main/src/library/Environment.flix

```diff
@@ -0,0 +1,136 @@
+    let iter = javaMap.entrySet().iterator();
+    let iterDone = () -> not iter.hasNext();
+    let iterNext = () -> {
+      let entry = iter.next() as ##java.util.Map$Entry;
+      let key = entry.getKey() as String;
+      let value = entry.getValue() as String;
+      (key, value)
+    };
+    Iterator.toMap(Iterator(iterDone, iterNext))
+
+  ///
+  /// Returns an map of the current system environment.
+  ///
+  pub def getEnv(): Map[String, String] =
+    import java.lang.System:getenv();
+    try {
+      let envMap = getenv();
+      _envMapToFlix(envMap)
+    } catch {
+      case _:##java.lang.Exception => Map.empty()
+    } as & Pure
+
+  ///
+  /// Gets the value of the specified environment variable.
+  ///
+  pub def getVar(name: String): Option[String] =
+    import java.lang.System:getenv(String);
+    try {
+      let ans = getenv(name) as & Pure;
+      if (Object.isNull(ans)) None else Some(ans)
+    } catch {
+      case _:##java.lang.Exception => None
+    }
+
+  ///
+  /// Gets the system property by name.
+  ///
+  pub def getProp(name: String): Option[String] =
+    import java.lang.System:getProperty(String);
+    try {
+      let ans = getProperty(name) as & Pure;
+      if (Object.isNull(ans)) None else Some(ans)
+    } catch {
+      case _:##java.lang.Exception => None
+    }
+
+  ///
+  /// Operating system name
+  ///
+  pub def getOsName(): Option[String] = getProp("os.name")
+
+  ///
+  /// Operating system architecture
+  ///
+  pub def getOsArch(): Option[String] = getProp("os.arch")
+
+  ///
+  /// Operating system version
+  ///
+  pub def getOsVersion(): Option[String] = getProp("os.version")
+
+  ///
+  /// File separator
+  ///
+  pub def getFileSeparator(): String =
+    import get java.io.File:separator as sep;
+    sep() as String & Pure
+
+  ///
+  /// Path separator
+  ///
+  pub def getPathSeparator(): String =
+    import get java.io.File:pathSeparator as pathSep;
+    pathSep() as String & Pure
+
+  ///
+  /// Line separator
+  ///
+  pub def getLineSeparator(): String =
+    import java.lang.System:lineSeparator();
+    lineSeparator() as String & Pure
+
+  ///
+  /// User's current working directory
+  ///
+  pub def getCurrentWorkingDirectory(): Option[String] = getProp("user.dir")
+
+  ///
+  /// Default temp file path
+  ///
+  pub def getTemporaryDirectory(): Option[String] = getProp("java.io.tmpdir")
+
+  ///
+  /// User's account name
+  ///
+  pub def getUserName(): Option[String] = getProp("user.name")
+
+  ///
+  /// User's home directory
+  ///
+  pub def getUserHomeDirectory(): Option[String] = getProp("user.home")
+
+  ///
+  /// Returns the number of virtual processors available to the JVM.
+  ///
+  /// The returned value is never less than one.
+  ///
+  pub def virtualProcessors(): Int32 =
+    import java.lang.Runtime:getRuntime();
+    import java.lang.Runtime.availableProcessors();
+    getRuntime().availableProcessors() as & Pure
+
+}
```

Comment:
Final comment is that we usually use 4 space indentation.

Reply:
It's done